src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -30,23 +30,27 @@ public class MeterProviderSdk
     {
         internal readonly ConcurrentDictionary<AggregatorStore, bool> AggregatorStores = new ConcurrentDictionary<AggregatorStore, bool>();
 
+        private readonly object collectLock = new object();
         private readonly CancellationTokenSource cts = new CancellationTokenSource();
         private readonly List<Task> collectorTasks = new List<Task>();
         private readonly MeterListener listener;
+        private readonly List<MeasurementProcessor> measurementProcessors = new List<MeasurementProcessor>();
+        private readonly List<MetricProcessor> metricProcessors = new List<MetricProcessor>();
```

Comment:
Inside the SDK it might make sense to have one single MetricProcessor, which can be a `CompositeProcessor` (similar like what we did for log/trace). Not a blocker though.

Reply:
yes! that's the plan. Will orchestrate it in such a way that one processor read the state, and provides it to the next ones.. Needs some non-trivial change, so making it in coming PRs.